src/main/java/com/google/sps/Trip.java

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Trip is the class for storing a single trip (could be multiple days).
+ */
+public class Trip {
+  // Trip cannot be longer than a month (31 days)
+  private static final int MAX_NUM_DAYS = 31;
+
+  private String tripName;
+  private LocalDate startDate;
+  private LocalDate endDate;
+  private int numDays;
+  private List<TripDay> tripDays;
+
+  /**
+   * Creates a new Trip.
+   *
+   * @param tripName The human-readable name for the trip. Must be non-null.
+   * @param startDate The start date for the trip. Must be non-null. Must be in yyyy-MM-dd date format.
+   * @param endDate The end date for the trip. Must be non-null. Must be in yyyy-MM-dd date format.
+   * @param tripDays The list of tripDays. Must be non-null.
+   */
+  public Trip(String tripName, String startDate, String endDate, List<TripDay> tripDays) {
+    if (tripName == null) {
+      throw new IllegalArgumentException("tripName cannot be null");
+    }
+
+    if (startDate == null) {
+      throw new IllegalArgumentException("startDate cannot be null");
+    }
+
+    if (endDate == null) {
+      throw new IllegalArgumentException("endDate cannot be null");
+    }
+
+    if (tripDays == null) {
+      throw new IllegalArgumentException("tripDays cannot be null. Use empty array instead.");
+    }
+
+    if (!isValidDate(startDate)) {
+      throw new IllegalArgumentException("Invalid startDate format. Must be in yyyy-MM-dd date format.");
+    }
+
+    if (!isValidDate(endDate)) {
```

Comment:
not sure you need these methods anymore, since you're doing the LocalDate.parse() below. Maybe have the method check *and* create the LocalDate? That way you aren't having to parse twice, and you don't have to have two try catches in the constructor